### PR TITLE
[WINDOWS] We are prepring the FileObserver for Windows OS.

### PR DESCRIPTION
**Pull request by pwielders**

### CODE REVIEW

File: Source/core/FileObserver.h [hunk 1]
```diff
@@ -245,42 +249,173 @@ class FileSystemMonitor : public Core::IResource {
...
+            }
         }
 
     public:
-        HANDLE Event() {
-            return(_syncEvent);
+        void Register(ICallback* client) {
+            ASSERT(std::find(_clients.begin(), _clients.end(), client) == _clients.end());
+
+            _clients.push_back(client);
```
Comment (MFransen69) [inline Source/core/FileObserver.h:310]: Not protected I guess for a reason? (performance and not used multithreaded)

File: Source/core/FileObserver.h [hunk 2]
```diff
@@ -245,42 +249,173 @@ class FileSystemMonitor : public Core::IResource {
...
+                    TRACE_L1("Could not start observing: %s", _filename.c_str());
+                }
+
+                for (auto& client : _clients) {
+                    client->Updated();
+                }
+            }
+        }
+
+        /*
```
Comment (MFransen69) [inline Source/core/FileObserver.h]: remove commented out code or do you still need it?

**Review by MFransen69 — CHANGES_REQUESTED** [review]

File: Source/core/FileObserver.h (hunk 1 above)
Comment (pwielders) [reply Source/core/FileObserver.h:310]: It is protected by the outer class. Inner class can not be directly accessed. So not required to lock (unless we want to double lock :-) )

File: Source/core/FileObserver.h (hunk 2 above)
Comment (pwielders) [reply Source/core/FileObserver.h]: Don't know if we still need it, found it with an example on internet. Wanted to store it for reference later on. Can also link the example from which it was taken :-)

**Comment by pwielders** [comment]
Eleborated a bit on the design decisions. Was a bit too much left to the reader :-) Hope it explains now.

**Review by MFransen69 — DISMISSED** [review]

**Review by MFransen69 — APPROVED** [review]

**Review by MFransen69 — APPROVED** [review]

**Pull request merged**